planning/route_handler/src/route_handler.cpp

```diff
@@ -1971,52 +1971,69 @@ bool RouteHandler::planPathLaneletsBetweenCheckpoints(
   lanelet::ConstLanelets * path_lanelets) const
 {
-  lanelet::Lanelet start_lanelet;
-  if (!lanelet::utils::query::getClosestLanelet(road_lanelets_, start_checkpoint, &start_lanelet)) {
+  lanelet::ConstLanelets start_lanelets;
+  if (!lanelet::utils::query::getCurrentLanelets(
+        road_lanelets_, start_checkpoint, &start_lanelets)) {
     return false;
   }
   lanelet::Lanelet goal_lanelet;
   if (!lanelet::utils::query::getClosestLanelet(road_lanelets_, goal_checkpoint, &goal_lanelet)) {
     return false;
   }
 
-  // get all possible lanes that can be used to reach goal (including all possible lane change)
-  const lanelet::Optional<lanelet::routing::Route> optional_route =
-    routing_graph_ptr_->getRoute(start_lanelet, goal_lanelet, 0);
-  if (!optional_route) {
-    RCLCPP_ERROR_STREAM(
-      logger_, "Failed to find a proper path!"
-                 << std::endl
-                 << "start checkpoint: " << toString(start_checkpoint) << std::endl
-                 << "goal checkpoint: " << toString(goal_checkpoint) << std::endl
-                 << "start lane id: " << start_lanelet.id() << std::endl
-                 << "goal lane id: " << goal_lanelet.id() << std::endl);
-    return false;
-  }
+  lanelet::Optional<lanelet::routing::Route> optional_route;
+  std::vector<lanelet::ConstLanelets> candidate_paths;
+  lanelet::routing::LaneletPath shortest_path;
+  bool is_route_found = false;
 
-  const lanelet::routing::LaneletPath shortest_path = optional_route->shortestPath();
   bool shortest_path_has_no_drivable_lane = hasNoDrivableLaneInPath(shortest_path);
   lanelet::routing::LaneletPath drivable_lane_path;
   bool drivable_lane_path_found = false;
 
-  if (shortest_path_has_no_drivable_lane) {
-    drivable_lane_path_found =
-      findDrivableLanePath(start_lanelet, goal_lanelet, drivable_lane_path);
-  }
+  for (const auto & start_lanelet : start_lanelets) {
+    optional_route = routing_graph_ptr_->getRoute(start_lanelet, goal_lanelet, 0);
+    if (!optional_route) {
+      RCLCPP_ERROR_STREAM(
+        logger_, "Failed to find a proper path!"
+                   << std::endl
+                   << "start checkpoint: " << toString(start_checkpoint) << std::endl
+                   << "goal checkpoint: " << toString(goal_checkpoint) << std::endl
+                   << "start lane id: " << start_lanelet.id() << std::endl
+                   << "goal lane id: " << goal_lanelet.id() << std::endl);
+    } else {
+      is_route_found = true;
 
-  lanelet::routing::LaneletPath path;
-  if (drivable_lane_path_found) {
-    path = drivable_lane_path;
-  } else {
-    path = shortest_path;
-  }
+      shortest_path = optional_route->shortestPath();
+      lanelet::ConstLanelets candidate_path_lanelets;
+      for (const auto & llt : shortest_path) {
+        candidate_path_lanelets.push_back(llt);
+      }
+      candidate_paths.push_back(candidate_path_lanelets);
+      if (shortest_path_has_no_drivable_lane) {
```

Comment:
Related to https://github.com/autowarefoundation/autoware.universe/pull/4013/files#r1255188461

I assume because previous shortest path checks will return false, then in that case, the next line might always be executed.